karate-core/src/main/java/com/intuit/karate/http/HttpClient.java

```diff
@@ -169,7 +169,7 @@ private T buildRequestInternal(HttpRequest request, ScriptContext context) {
                 buildCookie(cookie);
             }
         }
-        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method)) {
+        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method) || "DELETE".equals(method)) {
```

Comment:
Hi Peter,
Sorry but I have some questions here.
As I understand it, you are suggesting that we return null when body is null/isNull without throwing any exception.  If we do that, it makes body optional for all HTTP methods (including POST, PUT, PATCH and DELETE). Is my understanding correct? If yes, in my opinion POST, PUT or PATCH requests that do NOT have a body does not make any sense. Do we really want to do this?
l agree that in the Else case (body not null/isNull) we should call just call getEntityInternal(body, mediaType);

Thanks,
Satish